packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -193,44 +257,111 @@ async function initialize(editorInstance) {
  */
 function onSelectedTextChange() {
 	selectedText.value = editor.value?.getSelectedText() ?? '';
+	selectionRange.value =
+		!isEmpty(selectedText.value) && editor.value ? editor.value.getSelectionRange() : null;
+}
+
+function highlightDynamics() {
+	if (codeAsset.value?.files) {
+		const { name, files } = codeAsset.value;
+
+		Object.keys(files).forEach((fileName) => {
+			if (fileName === name) {
+				const { block } = files[fileName].dynamics;
+				// Loop through every highlighted block
+				for (let i = 0; i < block.length; i++) {
+					// Avoids rehighlighting
+					if (!existingMarkers.has(block[i])) {
+						// Extract start and end rows and highlight them in the editor
+						const match = block[i].match(/L(\d+)-L(\d+)/) || [];
+						const startRow = parseInt(match[1], 10) - 1;
+						const endRow = parseInt(match[2], 10) - 1;
```

Comment:
I think we can move this parsing into a helper function. 

Do you expect this to ever _not_ match the pattern? Because your fallback condition is not going to work for startRow/endRow

```
> const match = []
< undefined

> parseInt(match[2], 10)
< NaN
```

Reply:
I agree that this needs to be written/refactor, but could we just merge it for the demo, then create a new GitHub issue about it?